ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -1232,7 +1212,13 @@ private module ParameterNodes {
     final override Parameter getParameter() { none() }
 
     final override predicate isParameterOf(DataFlowCallable c, ParameterPosition pos) {
-      c = callable and pos.isSynthSplat()
+      c = callable and
+      exists(boolean hasActualSplat |
+        pos.isSynthSplat(hasActualSplat) and
+        if exists(TSynthSplatParameterShiftNode(c, _, _))
```

Comment:
I'm confused by this check. IIUC the shift node won't exist if the only splat parameter is at index 0. But should that count as not having an actual splat parameter?